### PR TITLE
Bugfix/v3.17.x.x start main executable with offline mode

### PR DESCRIPTION
**Pull request by Szczyrk**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/AppStarter.cs [hunk 1]
```diff
@@ -97,6 +100,10 @@ private void StartAppVersion(AppVersion? appVersion, string customArgs)
             {
                 appArgs += " " + appVersion.Value.MainExecutableArgs;
             }
+            else
+            {
+                appArgs += " " + _app.LocalMetaData.GetMainExecutableArgs();
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/AppStarter.cs:105]: What happens here if `_app.LocalMetaData.GetMainExecutableArgs()` will be a null?
This behavior is unclear. Please do `else if(string.IsNullOrEmpty(...))` and then assign.

File: Assets/PatchKit Patcher/Scripts/Patcher.cs [hunk 2]
```diff
@@ -899,6 +899,10 @@ private void ThreadUpdateApp(bool automatically, CancellationToken cancellationT
             {
                 _appInfo.Value = _app.RemoteMetaData.GetAppInfo(!automatically, _updateAppCancellationTokenSource.Token);
                 _remoteVersionId.Value = _app.GetLatestVersionId(!automatically, _updateAppCancellationTokenSource.Token);
+                var latestAppVersion =
```
Comment (genail) [inline Assets/PatchKit Patcher/Scripts/Patcher.cs:902]: Don't use `var` if the type cannot be determined by looking at the right side.

**Review by genail — CHANGES_REQUESTED** [review]

File: Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs [hunk 3]
```diff
@@ -40,6 +40,12 @@ private struct Data
             [DefaultValue("none")]
             [JsonProperty("product_key_encryption", DefaultValueHandling = DefaultValueHandling.Populate)]
             public string ProductKeyEncryption;
+            
+            [JsonProperty("main_executable")]
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs:44]: Should be `start_exe_path` to make it compatible with Electron launcher.

File: Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs [hunk 4]
```diff
@@ -40,6 +40,12 @@ private struct Data
...
+            public string MainExecutable;
+            
+            [JsonProperty("main_executable_args")]
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs:47]: Should be `start_exe_args` to make it compatible with Electron launcher.

File: Assets/PatchKit Patcher/Scripts/AppStarter.cs (hunk 1 above)
Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/AppStarter.cs:105]: I suggest introducing `mainExecutableArgs` just like in the function above.

File: Assets/PatchKit Patcher/Scripts/Patcher.cs [hunk 5]
```diff
@@ -899,6 +899,10 @@ private void ThreadUpdateApp(bool automatically, CancellationToken cancellationT
...
+                    _app.RemoteMetaData.GetAppVersionInfo(_remoteVersionId.Value.Value, false, cancellationToken);
+                _app.LocalMetaData.SetMainExecutableAndArgs(latestAppVersion.MainExecutable, latestAppVersion.MainExecutableArgs);
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/Patcher.cs:904]: I suggest doing it after a successful Update operation instead of at the start.

**Review by witcher112 — CHANGES_REQUESTED** [review]

File: Assets/PatchKit Patcher/Scripts/AppStarter.cs (hunk 1 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/AppStarter.cs]: done

File: Assets/PatchKit Patcher/Scripts/Patcher.cs (hunk 2 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs]: done

File: Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs (hunk 3 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs]: done

File: Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs (hunk 4 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs]: done

File: Assets/PatchKit Patcher/Scripts/Patcher.cs (hunk 5 above)
Comment (Szczyrk) [reply Assets/PatchKit Patcher/Scripts/Patcher.cs]: done

**Pull request merged**